pkg/provisioner/ironic/ironic.go

```diff
@@ -775,16 +723,7 @@ func (p *ironicProvisioner) UpdateHardwareState() (result provisioner.Result, er
 	return result, nil
 }
 
-func (p *ironicProvisioner) getUpdateOptsForNode(ironicNode *nodes.Node) (updates nodes.UpdateOpts, err error) {
-
-	hwProf, err := hardware.GetProfile(p.host.HardwareProfile())
-
-	if err != nil {
-		return updates, errors.Wrap(err,
-			fmt.Sprintf("Could not start provisioning with bad hardware profile %s",
-				p.host.HardwareProfile()))
-	}
-
+func (p *ironicProvisioner) getImageUpdateOptsForNode(ironicNode *nodes.Node, imageData *metal3v1alpha1.Image) (updates nodes.UpdateOpts, err error) {
 	// image_source
 	var op nodes.UpdateOp
 	if _, ok := ironicNode.InstanceInfo["image_source"]; !ok {
```

Comment:
At some point we probably want this check to compare the value, not just see if there is one. That should be another PR, though, since it has bigger implications.

Reply:
I have a patch for this but it needs tests: https://github.com/zaneb/baremetal-operator/commit/ed324acfcfb4d3efa77bba7527a575f16f8ca3a8
Feel free to borrow.